raiden/network/transport/matrix/transport.py

```diff
@@ -966,20 +1002,24 @@ def _get_room_for_address(self, address: Address, allow_missing_peers=False) ->
         room_is_empty = not bool(peer_ids & member_ids)
         if room_is_empty:
             last_ex: Optional[Exception] = None
-            retry_interval = 0.1
+            retry_interval = ROOM_JOIN_RETRY_INTERVAL
             self.log.debug(
-                "Waiting for peer to join from invite", peer_address=to_checksum_address(address)
+                "Waiting for peer to join from invite",
+                room=room,
+                peer_address=to_checksum_address(address),
             )
             for _ in range(JOIN_RETRIES):
                 try:
-                    member_ids = {member.user_id for member in room.get_joined_members()}
+                    member_ids = {
+                        member.user_id for member in room.get_joined_members(force_resync=True)
```

Comment:
This missing `force_resync=True` here was a nice subtle bug that caused a lot of the flakiness in the transport tests. 
Also shows that code duplication is always good for annoying errors